activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -127,6 +127,10 @@ def set_constraints(deferred, *constraints)
           def cancel_any_running_query
             return if @raw_connection.nil? || IDLE_TRANSACTION_STATUSES.include?(@raw_connection.transaction_status)
 
+            if PG.library_version >= 18_00_00 && Gem::Version.new(PG::VERSION) < Gem::Version.new("1.6.0")
+              raise NotImplementedError, "PG::Connection#cancel is not supported with the combination of pg gem #{PG::VERSION} and PostgreSQL libpq #{PG.library_version / 10000}. Please upgrade pg gem to 1.6.0 or later."
+            end
```

Comment:
Instead of raising, let's just not call `cancel` in this combination of versions. Make sure the reason for this conditional is documented in a comment.